TTS/tts/utils/speakers.py

```diff
@@ -155,15 +155,21 @@ def save_d_vectors_to_file(self, file_path: str) -> None:
         """
         self._save_json(file_path, self.d_vectors)
 
-    def set_d_vectors_from_file(self, file_path: str) -> None:
+    def set_d_vectors_from_file(self, file_path: str, data: List = None) -> None:
         """Load d_vectors from a json file.
 
         Args:
             file_path (str): Path to the target json file.
         """
         self.d_vectors = self._load_json(file_path)
-        speakers = sorted({x["name"] for x in self.d_vectors.values()})
-        self.speaker_ids = {name: i for i, name in enumerate(speakers)}
+
+        # load speakers from data, because during the training we can just use some speakers from d_vector_file
+        if data is not None:
+            self.speaker_ids, _ = self.parse_speakers_from_data(data)
+        else:
+            speakers = sorted({x["name"] for x in self.d_vectors.values()})
+            self.speaker_ids = {name: i for i, name in enumerate(speakers)}
+
```

Comment:
For training with d_vecs, speaker_manager listed all speakers present in speakers.json. But many times we only want to use some talkers (like when we use ununsed_speakers to ignore talkers). To ensure that during training only the speakers that will be used in training are printed, I modified this function.

Reply:
@Edresson if I'm fine-tuning a pre-trained VCTK model on a single new speaker, does this setting (speaker_ids) apply ?

Reply:
In this case, your new speaker will have samples in the training so this part of the code will not influence your training. This just counts the number of speakers and returns all speakers that are in the training dataset.